### DISCUSSION TITLE
Find-FalconDuplicate is giving a 0 KB file

### DISCUSSION THREAD

**Discussion by ghost** [CrowdStrike/psfalcon#267] [Q&A] [answered]
Hi guys,
I'm using psfalcon (2.2.3 as of this writing) to find duplicate hosts and hide them. The code I'm using below is no longer working :( I'm getting a blank 0 KB .csv file every time and it was working for a while and now it stopped. I rotated my api keys, same issue. I confirmed my permissions are correct (hosts read/write). What am I doing wrong?

```
Import-Module -Name PSFalcon
Request-FalconToken -Cloud us-2 -ClientId '' -ClientSecret ''
Find-FalconDuplicate | Export-Csv 'C:\'

$Duplicates = (Import-Csv 'C:\').device_id
Invoke-FalconHostAction -Name hide_host -Ids $Duplicates
```

**Comment by bk-cs** [comment, marked as answer]
`Find-FalconDuplicate` returns a warning message instead of standard output if no duplicates are found. Is that what you see if you try the command outside of your script? If you're only receiving the warning message, I would expect a blank CSV.

Adding an if statement will solve the script issue:
```powershell
Import-Module -Name PSFalcon
Request-FalconToken -Cloud us-2 -ClientId ... -ClientSecret ...
$Duplicate = Find-FalconDuplicate
if ($Duplicate) {
    $Timestamp = Get-Date -Format FileDate
    $Duplicate | Export-Csv "$(Join-Path $env:SystemDrive ('duplicates',$Timestamp -join '_')).csv" -NoTypeInformation -Append
    Invoke-FalconHostAction -Name hide_host -Id $Duplicate.device_id
}
```

**Comment by ghost** [comment]
@bk-cs Hi and thank you. I was able to figure out my root problem - my corporate firewall is blocking my API requests 🤡 so it didn't give me any warning messages. Please ignore this post entirely. Thank you for your help!
